packages/server/utils/atlassian/jiraImages.ts

```diff
@@ -0,0 +1,111 @@
+import cheerio from 'cheerio'
+import crypto from 'crypto'
+import fetch from 'node-fetch'
+import Redis from 'ioredis'
+import ms from 'ms'
+import getRedis from '../getRedis'
+import PROD from '../../PROD'
+
+export const NO_IMAGE_BUFFER = Buffer.from('X')
+export const IMAGE_TTL_MS = ms('2h')
+export const JIRA_IMAGES_ENDPOINT = '/jira-attachemenets'
+
+type UpdateJiraImagesResult = {
+  updatedDescription: string
+  imageUrlToHash: Map<string, string>
+}
+
+/**
+ * Parses a JIRA issue description and replaces the image urls with Parabol's image urls
+ * @param {string} descriptionHTML - HTML string of related Jira issue description
+ * @returns {UpdateJiraImagesResult} - Map of orginal image urls to hashed image urls and updated description
+ */
+export const updateJiraImageUrls = (
+  cloudId: string,
+  descriptionHTML: string
+): UpdateJiraImagesResult => {
+  const imageUrlToHash: Map<string, string> = new Map()
+  const projectBaseUrl = `https://api.atlassian.com/ex/jira/${cloudId}`
+
+  const $ = cheerio.load(descriptionHTML)
+  $('body')
+    .find('img')
+    .each((_i, img) => {
+      const imageUrl = $(img).attr('src')
+      if (!imageUrl) return
+
+      const absoluteImageUrl = `${projectBaseUrl}${imageUrl}`
+      const hashedImageUrl = createImageUrlHash(absoluteImageUrl)
+      imageUrlToHash.set(absoluteImageUrl, hashedImageUrl)
+
+      $(img).attr('src', createParabolImageUrl(hashedImageUrl))
+    })
+
+  return {updatedDescription: $.html(), imageUrlToHash}
+}
+
+export const downloadAndCacheImages = async (
+  authToken: string,
+  imageUrlToHash: Map<string, string>
+) => {
+  const redis = getRedis()
+  const promises: Promise<void>[] = []
+  for (const [imageUrl, hashedImageUrl] of imageUrlToHash) {
+    const cachedImage = await redis.get(hashedImageUrl)
+    if (cachedImage) return // skip if already cached
+
+    promises.push(downloadAndCacheImage(redis, authToken, hashedImageUrl, imageUrl))
+  }
+
+  await Promise.all(promises)
+}
+
+const createParabolImageUrl = (hashedImageUrl: string): string => {
+  //locally API lives on different port than the frontend application
+  return `${
+    !PROD ? `http://localhost:${process.env.SOCKET_PORT}` : ''
+  }${JIRA_IMAGES_ENDPOINT}/${hashedImageUrl}`
+}
+
+const downloadAndCacheImage = async (
+  redis: Redis.Redis,
+  authToken: string,
+  hashedImageUrl: string,
+  imageUrl: string
+) => {
+  try {
+    redis.setBuffer(hashedImageUrl, NO_IMAGE_BUFFER)
+    const imageBuffer = await fetchImage(authToken, imageUrl)
+    if (!imageBuffer) return
+    redis.setBuffer(hashedImageUrl, imageBuffer, 'PX', IMAGE_TTL_MS)
+  } catch (error) {
+    console.error(error)
+  }
+}
+
+const fetchImage = async (authToken: string, url: string): Promise<Buffer | null> => {
+  const response = await fetch(url, {
```

Comment:
fetch needs a timeout & abort controller.
consider moving this logic into the AtlassianServerManager so it can use fetchWithTimeout

Reply:
When I add a method that needs to return a `Buffer` to `AtlassianManager` it complains the `response.buffer()` doesn't exist on type `fetch`. It's most probably because `AtlassianManager` has `abstract fetch: typeof fetch` which uses libdom type. 

I see the following solutions
1. extract `fetchWithTimeout` as a separate function that'd have to be bound to a proper `fetch` 
2. instead of `node-fetch` and native browser fetch use something like this https://github.com/matthew-andrews/isomorphic-fetch 
3. Simply copy-paste `fetchWithTimeout` as a separate function that uses `node-fetch` 

Unless there's some simpler solution that'd require changing types in `AtlassianManager` and `AtlassianServerManager`. Any thoughts @mattkrick? 🙏

Reply:
buffer doesn't exist, but arrayBuffer does :wink: 

see https://github.com/ParabolInc/parabol/blob/de68ce6722085bb6ca0ac9b27a08733188dc2b6c/packages/client/utils/AtlassianManager.ts#L438

Reply:
I wanted to avoid the conversion ;) will move the logic to atlassian manager then!

Reply:
Now I realized I'm not handling `content-type` correctly and I always assume it's `png` 😭 
@mattkrick do you have any thoughts about how to store content types along with buffers in Redis?
What about storing it as a string with key like `content-type|imageHash` and just retrieve it endpoint handler and return the correct value? 🤔

Reply:
here's what i'd do:
- the jira URL is /:fileName.[ext]
- remove the .[ext] before hashing the fileName
- then put the extension on the hashed fileName
- content-type can be determined from the extension using the package mime-types https://github.com/ParabolInc/parabol/blob/8c68bf771a350b31c7948e6bd11c5c110e30c977/packages/server/utils/StaticServer.ts#L2

Reply:
that's a great idea! 🙏   added.